Cargo.toml

```diff
@@ -0,0 +1,6 @@
+[workspace]
```

Comment:
Do you need a workspace here? Are multiple crates expected?

Reply:
I am planning to develop lib and bin as different crates, not sure if it's a good practice

Reply:
Actually nvm lets make it one crate